### ISSUE TITLE
org-extra-emphasis-alist markers cancel out evil-surround functionality

### ISSUE THREAD

**Issue by qaptoR** [kjambunathan/org-extra-emphasis#11]
I've chosen my alist markers to be `!`, `$`, `%`, `&`, `^`

They work perfectly well as per this package, but when trying to change the characters using evil-surround, it does not work. I get message `no such pair found`.

I did open an issue on the evil-surround repo, but my instinct tells me it is a bug with this package because the issue is only present with the symbols defined by `org-extra-emphasis-alist`.

I can look into it myself, but before I do I'd like to find out if anyone more familiar with this package has an idea of what could be causing the trouble

**Comment by kjambunathan** [comment]
Add this to your `.emacs`

```lisp
(use-package emacs
  :config (setq my-org-extra-emphasis-markers
	    '("!" "$" "%" "&" "^")))

(custom-set-variables
 `(org-extra-emphasis-alist
   ',(thread-last my-org-extra-emphasis-markers
		  (seq-map-indexed
		   (lambda (it i)
		     (list it (intern (format "org-extra-emphasis-%02d" (+ 1 i)))))))))

(custom-set-faces
 '(org-extra-emphasis-01 ((t (:foreground "red" :weight bold))))
 '(org-extra-emphasis-02 ((t (:background "green" :weight bold))))
 '(org-extra-emphasis-03 ((t (:foreground "blue" :weight bold))))
 '(org-extra-emphasis-04
   ((t
     (:inherit org-extra-emphasis :background "pale goldenrod"
	       :underline t))))
 '(org-extra-emphasis-05
   ((t
     (:inherit org-extra-emphasis :background "pale goldenrod"
	       :underline t)))))

(with-eval-after-load 'evil-surround
  (thread-last my-org-extra-emphasis-markers
	       (seq-do
		(lambda (marker)
		  (add-to-list 'evil-surround-pairs-alist
			       (list
				(string-to-char marker)
				(cons marker marker)))))))
```

**Comment by kjambunathan** [comment]
This is another config equivalent to the one I shared previously

```lisp

(use-package emacs
  :config (defvar my-org-extra-emphasis-markers
	    '("!" "$" "%" "&" "^")))

(use-package org-extra-emphasis
  :custom
  (org-extra-emphasis-alist
   '(("!" org-extra-emphasis-01)
     ("$" org-extra-emphasis-02)
     ("%" org-extra-emphasis-03)
     ("&" org-extra-emphasis-04)
     ("^" org-extra-emphasis-05)
     ))
  :custom-face
  (org-extra-emphasis-01 ((t (:foreground "red" :weight bold))))
  (org-extra-emphasis-02 ((t (:background "green" :weight bold))))
  (org-extra-emphasis-03 ((t (:foreground "blue" :weight bold))))
  (org-extra-emphasis-04
   ((t
     (:inherit org-extra-emphasis :background "yellow"
	       :underline t))))
  (org-extra-emphasis-05
   ((t
     (:inherit org-extra-emphasis :foreground "orange"
	       :underline t)))))

(use-package evil-surround
  :config
  (progn
    (add-to-list 'evil-surround-pairs-alist '(33 ("!" . "!")))
    (add-to-list 'evil-surround-pairs-alist '(36 ("$" . "$")))
    (add-to-list 'evil-surround-pairs-alist '(37 ("%" . "%")))
    (add-to-list 'evil-surround-pairs-alist '(38 ("&" . "&")))
    (add-to-list 'evil-surround-pairs-alist '(94 ("^" . "^")))))
```

**Comment by kjambunathan** [comment]
If you use `$` (or any other markers used by `org`) then you may run in to issues during export to HTML.  Just adding a remark so that you don't get surprised by inconsistent results during export.

```
- !org-extra-emphasis-01!
- $org-extra-emphasis-02$
- %org-extra-emphasis-03%
- &org-extra-emphasis-04&
- ^org-extra-emphasis-05^

```
![org-extra-emphasis-issue-11](https://github.com/QiangF/org-extra-emphasis/assets/19869864/5d82c176-0c47-4901-9487-878abb91ca9b)

**Comment by kjambunathan** [comment]
I am not familiar with `vim` or `evil`, so if the above configuration doesn't work you have to tell me

- what config you are using
- how to do `evil-surround`

You can also do `M-x toggle-debug-on-error` and share what `Emacs` shows you in `*Backtrace*` buffer.

**Comment by kjambunathan** [comment]
> I did open an issue on the evil-surround repo,

[**_evil-surround issue#206_**: org-extra-emphasis-alist markers return 'no such pair found' ](https://github.com/emacs-evil/evil-surround/issues/206)

**Comment by qaptoR** [comment]
Hey, thank-you for getting back about this, I will try these config solutions and edit this comment with the result as soon as I get the chance.

One thing I have learned since posting this issue, is that evil-surround doesn't work for just ANY pair of characters, which I thought it did when I posted, which is why I thought the issue was with this repo. It could still be that your configs solve the issue anyway because I see that you are updating a list for evil surround. But it definitely isn't a bug otherwise for evil-surround to just NOT know about the extra-emphasis markers.

Also, thanks for the heads up about the '$' notation possibly causing an issue for export. I don't think it will be much of a problem because I'm very far from even thinking about exporting my notes to another format, So I will cross that bridge when it comes to it.
But, your package does FANTASTIC at creating the extra markers, and having them read properly. I actually use all of these additional markers, `%, $, ^, &, &&, ^^, $$, !!, %%, //, ==, ~~, __` where the `##` ones apply a background color for what the font color would otherwise be. I was very impressed that it could differentiate between `/` and `//` so easily. So again, great job.
